ghmirror/data_structures/redis_data_structures.py

```diff
@@ -21,44 +21,55 @@
 import redis
 
 
+REDIS_ENDPOINT = os.environ.get('REDIS_ENDPOINT')
```

Comment:
I think the 2 endpoints should come from evironment variables. Something like:
```
PRIMARY_ENDPOINT = os.environ.get('PRIMARY_ENDPOINT')
READER_ENDPOINT = os.environ.get('READER_ENDPOINT', PRIMARY_ENDPOINT)
```
Too bad that terraform does not expose that reader endpoint, but if we want to infer it (and include in the secret), the right place to do that would be here: https://github.com/app-sre/qontract-reconcile/blob/master/utils/terrascript_client.py#L1031-L1043
Otherwise we can also set them as parameters in the saas file.

Reply:
I like the first option more. setting them as parameters in the saas file would involve some hardcoding of endpoint names no?

it seems the terraform client script is handling all elasticache instances and there will not always be two endpoints such as when someone is running elasticache in cluster mode. So i guess we can leave the replica endpoint blank if it doesn't exist?

Reply:
Yes, think of the saas file as "deployment configuration", so it's ok to manually write there the values we want for deployment and to manually update them when it's needed. The domains won't change unless we recreate the redis instance, so we would already be "hands on".

About the terraform, yes, sounds like a plan. It would also be ok if we don't create the replica endpoint at all on cluster mode. It's up to you. Btw, I like this terraform approach because everybody else will benefit from it.